src/main/java/org/jboss/msc/service/ServiceContainer.java

```diff
@@ -247,7 +263,24 @@ public static ServiceContainer create(int coreSize, long keepAliveTime, TimeUnit
          * @return a new service container instance
          */
         public static ServiceContainer create(String name, int coreSize, long keepAliveTime, TimeUnit keepAliveTimeUnit, boolean autoShutdown) {
-            return new ServiceContainerImpl(name, calculateCoreSize(coreSize), keepAliveTime, keepAliveTimeUnit, autoShutdown);
+            return new ServiceContainerImpl(name, calculateCoreSize(coreSize), MAX_THREADS_COUNT, keepAliveTime, keepAliveTimeUnit, autoShutdown);
+        }
+
+
+        /**
+         * Create a new instance with a given name and specified initial thread pool settings.
+         *
+         * @param name the name of the new container
+         * @param coreSize the core pool size (must be greater than zero)
+         * @param maxSize the maximum pool size (must be greater than zero)
+         * @param keepAliveTime the amount of time that non-core threads should linger without tasks
+         * @param keepAliveTimeUnit the time unit for {@code keepAliveTime}
+         * @param autoShutdown {@code true} to automatically shut down the container at VM exit, {@code false} otherwise
+         * @return a new service container instance
+         */
+        public static ServiceContainer create(String name, int coreSize, int maxSize, long keepAliveTime, TimeUnit keepAliveTimeUnit, boolean autoShutdown) {
+            final int realCoreSize = calculateCoreSize(coreSize);
+            return new ServiceContainerImpl(name, realCoreSize, Math.max(realCoreSize, Math.min(maxSize, MAX_THREADS_COUNT)), keepAliveTime, keepAliveTimeUnit, autoShutdown);
```

Comment:
Perhaps some DEBUG logging if maxSize >  MAX_THREADS_COUNT, since that means the setting from the caller is being ignored in favor of what typically is calculateCoreSize() << 3 ?

I admit this is a nit as that seems like a corner case.

Reply:
TBH I am somewhat inclined to get rid of these properties completely, since Jason's correct CPU count calculation code was added in wildfly-common.  Any code which really cares about the thread pool size could just specify it and there's no reason why we couldn't just accept it.

Reply:
A WildFly Core based server sets the pool size itself based on its own calculations, so jboss.msc.max.container.threads is not relevant there.

A HostController however just does ServiceContainer.Factory.create("host-controller", false) so it's *possible* that someone was using that property as a configuration hook. Really unlikely IMO.

Reply:
That configuration property was added because of MSC-144.

Reply:
I think this current code breaks MSC-144 then, right? As without a user setting change the max is now Math.max(thecalculatedcoresize, theconfiguredmaxsize).